pkg/cmd/dev/builder.go

```diff
@@ -124,6 +124,19 @@ func (d *dev) getDockerRunArgs(
 	} else {
 		args = append(args, "-i")
 	}
+	gitDir, err := d.exec.CommandContextSilent(ctx, "git", "rev-parse", "--git-dir")
+	if err != nil {
+		return nil, err
+	}
+	gitCommonDir, err := d.exec.CommandContextSilent(ctx, "git", "rev-parse", "--git-common-dir")
+	if err != nil {
+		return nil, err
+	}
+	// If run from inside a git worktree
+	if string(gitDir) != string(gitCommonDir) {
+		mountPath := strings.TrimSuffix(strings.TrimSpace(string(gitCommonDir)), "/.git")
```

Comment:
Let's mount the `.git` directory, not the checkout. i.e. don't strip out the `/.git` thing. The rest of the repository directory (the checkout) is not necessary for doing `git status`.